ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_linalg.py

```diff
@@ -589,6 +589,69 @@ def test_cond(
     )
 
 
+@st.composite
+def _generate_solve_triangular_dtype_and_arrays(draw):
+    input_dtype = [draw(st.sampled_from(draw(helpers.get_dtypes("numeric"))))]
+    matrices_dims = draw(
+        st.lists(st.integers(min_value=2, max_value=10), min_size=2, max_size=2)
+    )
+    shape_1 = (matrices_dims[0], matrices_dims[0])
+
+    matrix_1 = draw(
+        helpers.dtype_and_values(
+            shape=shape_1,
+            dtype=input_dtype,
+            min_value=-10,
+            max_value=10,
+        )
+    )
+    matrix_2 = draw(
+        helpers.dtype_and_values(
+            shape=shape_1,
+            dtype=input_dtype,
+            min_value=-10,
+            max_value=10,
+        )
+    )
+
+    return input_dtype, [matrix_1[1][0], matrix_2[1][0]]
```

Comment:
I feel we don't need this function here, you can simply use `dtype_and_values` and pass `num_arrays=2`. You can also use these keyword arguments for generating the shape values - 

``` 
min_num_dims
max_num_dims
min_dim_size
max_dim_size
```